error.go

```diff
@@ -74,9 +74,16 @@ func isBadConn(err error, allowTimeout bool) bool {
 	}
 
 	if isRedisError(err) {
-		// Close connections in read only state in case domain addr is used
-		// and domain resolves to a different Redis Server. See #790.
-		return isReadOnlyError(err)
+		if isReadOnlyError(err) {
+			// Close connections in read only state in case domain addr is used
+			// and domain resolves to a different Redis Server. See #790.
+			return true
+		} else if isMovedSameConnAddr(err, addr) {
+			// Close connections when we are asked to move to the same addr
+			// of the connection. Force a DNS resolution when all connections
+			// of the pool are recycled
+			return true
+		}
```

Comment:
Replace `else if` with `else` and add `return false` as a final clause.

Reply:
Moved to a switch, hope that this also fixes the broken tests.